docker-compose.yml

```diff
@@ -1,29 +1,56 @@
 version: '3.7'
 
+x-shared_environment: &shared_environment
+  LOG_LEVEL: ${LOG_LEVEL}
+  DATABASE_HOST: db
+  DATABASE_NAME: ${DATABASE_NAME:-vapor_database}
```

Comment:
What does this syntax do? Does that take `DATABASE_NAME` from the env or use `vapor_database`? 
If so, how you pass environment flags into docker? That could be a nice addition to the comments section below.

Reply:
> Does that take `DATABASE_NAME` from the env or use "vapor_database"?

Correct. So, it takes it from the environment where the docker-compose file is interpreted. If you run `docker-compose ...` or `docker stack deploy -c docker-compose.yml ...` then it's the environment running those commands. This can be a nice way to give stuff some testing defaults while also allowing your build-box to have production versions of these variables set from some private ENV var store, perhaps a .environment file stored in AWS.

If you want to set them locally, you have the usual suspects:
1. Set them for your shell session: `export DATABASE_USERNAME=...`, etc.
2. Set them for a single command: `DATABASE_USERNAME=... docker-compose up`.
3. Set them in a new shell script that in turn executes the relevant docker commands.

I'll make a note in the comments section.